lib/pf/api/queue_cluster.pm

```diff
@@ -0,0 +1,285 @@
+package pf::api::queue_cluster;
+
+=head1 NAME
+
+pf::api::queue_cluster -
+
+=cut
+
+=head1 DESCRIPTION
+
+pf::api::queue_cluster
+
+=cut
+
+use strict;
+use warnings;
+use pf::cluster qw($cluster_enabled);
+use pf::log;
+use Moo;
+use List::Util qw(shuffle);
+use CHI;
+our $CHI_CACHE = CHI->new(driver => 'RawMemory', datastore => {});
+
+=head2 queue
+
+The queue to submit to
+
+=cut
+
+has queue => (
+    is      => 'rw',
+    default => 'general',
+);
+
+=head2 local_client
+
+The local queue client
+
+=cut
+
+has local_client => (
+    is => 'rw',
+    builder => 1,
+    lazy => 1,
+);
+
+
+=head2 server_recheck_timeout
+
+server_recheck_timeout
+
+=cut
+
+has server_recheck_timeout => (
+    is => 'rw',
+    default => 60,
+);
+
+=head2 jsonrpc_args
+
+jsonrpc_args
+
+=cut
+
+has jsonrpc_args => (is => 'rw', default => sub { {} }) ;
+
+=head2 _build_local_client
+
+Build the local client
+
+=cut
+
+sub _build_local_client {
+    my ($self) = @_;
+    require pf::pfqueue::producer::redis;
```

Comment:
Any reason why you aren't "using" the module ? That would be more effective I believe